src/types/Pattern.ts

```diff
@@ -171,6 +171,10 @@ export type ObjectLiteralPattern<a> =
     }
   | never;
 
+export type ObjectExactPattern<a> = {
+  readonly [k in keyof a]: Pattern<a[k]>;
+};
```

Comment:
Is it to force people to include any key that exist's on an object? I think that makes a lot of sense, but how does it behave when the input has type `unknown` and `any`? Could you add tests for that?